### PR TITLE
Feature deploy app

### PR DESCRIPTION
**Pull request by Dominic-go**
## Ticket
- [Deploy app](https://www.notion.so/Deploy-app-05b82f08502246d28d6cd663088c2cc1?pvs=4)

## Task
- [x]  Internal testing (play console)
- [x]  Firebase distribution
- [x]  Fastlane
- [x]  Github Action

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+          java-version: "17.0"
+
+      - name: Setup Flutter
+        uses: subosito/flutter-action@v2
+        with:
+          channel: 'stable'
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: '3.2'
+      - name: Install bundle
```
Comment (drake-ho-goldenowl) [inline .github/workflows/release.yml]: Replace to
  ```
        - name: Install Fastlane
        run: gem install fastlane
 ```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+          KEYSTORE_STORE_PASSWORD: ${{ secrets.KEYSTORE_STORE_PASSWORD }}
+
+      - name: Create Google Play Config file
+        run : |
+          echo "$PLAY_CONFIG_JSON" > fastlane_secret_key.json.json.b64
+          base64 -d -i fastlane_secret_key.json.json.b64 > fastlane_secret_key.json.json
+        env:
+          PLAY_CONFIG_JSON: ${{ secrets.PLAY_CONFIG_JSON }}
+      - name: Distribute app to Beta track 🚀
+        run: bundle exec fastlane beta
```
Comment (drake-ho-goldenowl) [inline .github/workflows/release.yml]: Change to
```
fastlane beta
```
Rememeber set `working-directory`

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    branches: ["dev", "feature/depoy"]
+    paths: 
+      - ".github/workflows/release.yml"
+
+jobs:
+  distribute:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
```
Comment (drake-ho-goldenowl) [inline .github/workflows/release.yml]: Change to 
```
actions/checkout@v2
```

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -26,13 +26,8 @@ jobs:
         uses: subosito/flutter-action@v2
         with:
           channel: 'stable'
-      - uses: ruby/setup-ruby@v1
-        with:
-          ruby-version: '3.2'
```
Comment (drake-ho-goldenowl) [inline .github/workflows/release.yml]: Sorry! Please keep this line code

File: .github/workflows/release.yml (hunk 2 above)
Comment (drake-ho-goldenowl) [reply .github/workflows/release.yml]: Please pay attention to the line 'Remember'

Comment (Dominic-go) [reply .github/workflows/release.yml]: Oh I'm sorry. I'll set working-directory

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+          KEYSTORE_STORE_PASSWORD: ${{ secrets.KEYSTORE_STORE_PASSWORD }}
+
+      - name: Create Google Play Config file
+        run : |
+          echo "$PLAY_CONFIG_JSON" > fastlane_secret_key.json.json.b64
+          base64 -d -i fastlane_secret_key.json.json.b64 > fastlane_secret_key.json.json
+        env:
+          PLAY_CONFIG_JSON: ${{ secrets.PLAY_CONFIG_JSON }}
+      - name: Distribute app to Beta track 🚀
+        working-directory: gallery_app/android
```
Comment (drake-ho-goldenowl) [inline .github/workflows/release.yml]: add line
```
shell: pwsh
```

File: gallery_app/android/fastlane/Fastfile [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+default_platform(:android)
+
+platform :android do
+  desc "Runs all the tests"
+  lane :distribute do
+      build_android_app(
+          task: "assembg",
+          build_type: "Release"
+      )
```
Comment (drake-ho-goldenowl) [inline gallery_app/android/fastlane/Fastfile]: Please use 'sh'

**Review by drake-ho-goldenowl — APPROVED** [review]

**Pull request merged**